packages/interface-adapter/lib/adapter/types.ts

```diff
@@ -18,6 +19,18 @@ export type BlockType = EvmBlockType | any;
 export type Transaction = EvmTransaction | any;
 export type TransactionReceipt = EvmTransactionReceipt | any;
 export type TxHash = string;
+export type TransactionReportData = {
+  timestamp: number;
+  from: string;
+  balance: string;
+  gasUnit: string;
+  gasPrice: string;
+  gas: BN;
+  mainUnit: string;
+  value: string;
+  cost: BN;
+  convertCost: (val: BN) => string;
```

Comment:
Naming-wise, `convertCost()` is a bit implicit... convert to what? How about `displayCost()` instead?

Separately, what about moving this function to the interface adapter itself? e.g. `interfaceAdapter.displayCost()`. Maybe the name could be better, but this function won't change per-transaction report, right?

Reply:
The reason I did it this way is because `convertCost` is not only called on `postDeploy`, but also on `postMigrate`. `postDeploy` has the contract as part of the data to call the interface adapter, but `postMigrate` doesn't.

It can be refactor so that `postMigrate` takes an object with the `interfaceAdapter` as well as `isLast` parameter (which I suppose it's possible since that occurs in `@truffle/migrate`).

Reply:
Let's just pass the same data to `postMigrate` as we do to `postDeploy`. I prefer that instead of this function here.

Reply:
I've refactored `postMigrate` to take a data object like other events. The payload is not the same as `postDeploy` as only the `interfaceAdapter` is needed and we don't have access to everything else, but I've been able to get rid of `convertCost` from the report and now `displayCost` is an `interfaceAdapter` method.